core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/PaginatorUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.util;
+
+import java.util.Collection;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@SdkInternalApi
+public final class PaginatorUtils {
+
+    private PaginatorUtils() {
+    }
+
+
+    /**
+     * Checks if the output token is available.
+     *
+     * @param outputToken the output token to check
+     * @param <T> the type of the output token
+     * @return true if the output token is non-null or non-empty if the output token is a map or Collection type
+     */
+    public static <T> boolean isOutputTokenAvailable(T outputToken) {
+        if (outputToken == null) {
+            return false;
+        }
+
+        if (outputToken instanceof Map) {
+            return !((Map) outputToken).isEmpty();
+        }
+
+        if (outputToken instanceof Collection) {
+            return !((Collection) outputToken).isEmpty();
+        }
+
+        return true;
```

Comment:
nit: you can remove line 37-39 and replace this with `outputToken != null`

Reply:
I like the current way instead of nested if statements. (my personal opinion)

Reply:
Hmm, I kind of like the current way. We can reduce 2 line of code with what you suggested.

```java
    public static <T> boolean isOutputTokenAvailable(T outputToken) {
//        if (outputToken == null) {
//            return false;
//        }

        if (outputToken instanceof Map) {
            return !((Map) outputToken).isEmpty();
        }

        if (outputToken instanceof Collection) {
            return !((Collection) outputToken).isEmpty();
        }

        return outputToken != null;
    }
```

But it seems more intuitive to me with the current way where we check null first and don't need to follow the logic all the way down if it's null.

Reply:
Yup, it's fine with me :)